core.d.ts

```diff
@@ -1,4 +1,9 @@
-import type {Readable as ReadableStream} from 'node:stream';
+/**
+ * Typings for primary entry point, Node.js specific typings can be found in index.d.ts
+ */
+
+import type {ReadableStream as WebReadableStream} from 'node:stream/web';
```

Comment:
This should be available globally.

Reply:
The global type in incompatible with Node.js type. If I change it here, the problem will appear elsewhere,

Reply:
I'm not sure what that means, but this is `core.d.ts`, so it shouldn't import types only available for Node.js.

Reply:
The Node.js Readable types are incompatible with the global `lib.dom.d.ts` types, one things the challenges mentioned in https://github.com/sindresorhus/file-type/pull/588#issuecomment-1493880124, related to this issue https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60377, which unfortunately get closed with PR aiming to resolve that.

I used the Node.js, as Node.js has been the primary supported platform. Using the `lib.dom.d.ts` I need hack in types mapping cast somewhere.

Maybe good accept both, an do an ugly type cast, just for the convenience for our users.

Off-topic: The other mind f*ck, was the [BYOB Readable Stream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamBYOBReader), The "Bring-Your-Own-Buffer" reader:
> defines a reader for a [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream) that supports **zero-copy** reading from an underlying byte source

Well the first things the **zero-copy** method does is hijacking _the buffer_ you bring, and essentially turns into junk (there is formal property for this state, forgot the name), meaning it can no longer be used. Which essentially forces you to create a new Buffer (as it becomes totally useless after providing it), and then copying the data to the Buffer you wanted to have it written in the first place. So the only feature BYOB has, versus the [ReadableStreamDefaultReader](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamDefaultReader) is that you can control the chunk length to be written. How confusing, what a disappointment.